### ISSUE TITLE
EnrichedActivity Not Returning Reactions

### ISSUE THREAD

**Issue by rlee1990** [GetStream/stream-feed-flutter#61]
I am not sure what is happening. I add a reaction like so:
```
widget.streamClient.reactions
                      .add('like', widget.post.id,
                          userId: widget.streamClient.currentUser.userId,
                          data: {
                            'value': index,
                          });
```
Then I get the enriched activities like so:
```
 timeline
        .getEnrichedActivities(
            limit: 100,
            offset: 0,
            flags: EnrichmentFlags()
                .withReactionCounts()
                .withOwnReactions()
                .withRecentReactions())
```
It will return the activities but not any reactions it's like they do not get updated even though I get a return value.

```
I/flutter (29491): *** Request ***
I/flutter (29491): uri: https://api.stream-io-api.com/api/v1.0/reaction/?api_key=hmwext26344q&location=unspecified
I/flutter (29491): method: POST
I/flutter (29491): responseType: ResponseType.json
I/flutter (29491): followRedirects: true
I/flutter (29491): connectTimeout: 10000
I/flutter (29491): sendTimeout: 0
I/flutter (29491): receiveTimeout: 10000
I/flutter (29491): receiveDataWhenStatusError: true
I/flutter (29491): extra: {}
I/flutter (29491): headers:
I/flutter (29491):  stream-auth-type: jwt
I/flutter (29491):  x-stream-client: stream-feed-dart-client-android-0.1.2
I/flutter (29491):  Authorization: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoiMEFKWmZVT1Rxb1hMOHlCeVhZODZUYzBMS3VsMnBlcnNvbmFsIn0.YnnMv2ApNpKPdh6aBCQm9hlB-W7C0DDyuzAMO2g61IA
I/flutter (29491):  content-type: application/json; charset=utf-8
I/flutter (29491): data:
I/flutter (29491): Reaction(null, like, bc1118f0-b2d2-11eb-8c76-1254437355c5, 0AJZfUOTqoXL8yByXY86Tc0LKul2personal, null, null, null, null, null, null, {value: 4}, null, null)
I/flutter (29491):
I/flutter (29491): *** Response ***
I/flutter (29491): uri: https://api.stream-io-api.com/api/v1.0/reaction/?api_key=hmwext26344q&location=unspecified
I/flutter (29491): statusCode: 201
I/flutter (29491): headers:
I/flutter (29491):  connection: keep-alive
I/flutter (29491):  cache-control: no-cache
I/flutter (29491):  date: Wed, 12 May 2021 19:14:05 GMT
I/flutter (29491):  access-control-allow-origin: *
I/flutter (29491):  x-ratelimit-limit: 1000
I/flutter (29491):  x-ratelimit-reset: 1620846900
I/flutter (29491):  x-ratelimit-remaining: 999
I/flutter (29491):  content-type: application/json;charset=utf-8
I/flutter (29491):  access-control-max-age: 86400
I/flutter (29491):  server: nginx
I/flutter (29491):  access-control-allow-headers: x-requested-with, content-type, accept, origin, authorization, x-csrftoken, x-stream-client, stream-auth-type
I/flutter (29491):  access-control-allow-methods: GET, POST, PUT, PATCH, DELETE, OPTIONS
I/flutter (29491):  content-length: 691
I/flutter (29491): Response Text:
I/flutter (29491): {"created_at":"2021-05-12T19:14:05.323360Z","updated_at":"2021-05-12T19:14:05.323360Z","id":"e95a85d0-b44b-4017-9fcd-cfae75d62d90","user_id":"0AJZfUOTqoXL8yByXY86Tc0LKul2personal","user":{"created_at":"2021-04-13T23:04:05.306176Z","updated_at":"2021-04-13T23:04:05.306176Z","id":"0AJZfUOTqoXL8yByXY86Tc0LKul2personal","data":{"gender":"Female","name":"jane doe","photo":"https://firebasestorage.googleapis.com/v0/b/fire-snab.appspot.com/o/profile-image-placeholder.png?alt=media&token=b17598bb-a510-4167-8354-ab75642ba89e"}},"kind":"like","activity_id":"bc1118f0-b2d2-11eb-8c76-1254437355c5","data":{"value":4},"parent":"","latest_children":{},"children_counts":{},"duration":"5.79ms"}
I/flutter (29491):
I/flutter (29491): Reaction(e95a85d0-b44b-4017-9fcd-cfae75d62d90, like, bc1118f0-b2d2-11eb-8c76-1254437355c5, 0AJZfUOTqoXL8yByXY86Tc0LKul2personal, , 2021-05-12 19:14:05.323360Z, 2021-05-12 19:14:05.323360Z, null, {created_at: 2021-04-13T23:04:05.306176Z, updated_at: 2021-04-13T23:04:05.306176Z, id: 0AJZfUOTqoXL8yByXY86Tc0LKul2personal, data: {gender: Female, name: jane doe, photo: https://firebasestorage.googleapis.com/v0/b/fire-snab.appspot.com/o/profile-image-placeholder.png?alt=media&token=b17598bb-a510-4167-8354-ab75642ba89e}}, null, {value: 4}, {}, {})
```

**Comment by sachaarbonel** [comment]
Hey @rlee1990 can you link the getEnrichedActivity response log?

**Comment by rlee1990** [comment]
@sachaarbonel here it is:
```
I/flutter (29491): *** Request ***
I/flutter (29491): uri: https://api.stream-io-api.com/api/v1.0/enrich/feed/timeline/0AJZfUOTqoXL8yByXY86Tc0LKul2personal/?limit=100&offset=0&api_key=hmwext26344q&location=unspecified
I/flutter (29491): method: GET
I/flutter (29491): responseType: ResponseType.json
I/flutter (29491): followRedirects: true
I/flutter (29491): connectTimeout: 10000
I/flutter (29491): sendTimeout: 0
I/flutter (29491): receiveTimeout: 10000
I/flutter (29491): receiveDataWhenStatusError: true
I/flutter (29491): extra: {}
I/flutter (29491): headers:
I/flutter (29491):  stream-auth-type: jwt
I/flutter (29491):  x-stream-client: stream-feed-dart-client-android-0.1.2
I/flutter (29491):  Authorization: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoiMEFKWmZVT1Rxb1hMOHlCeVhZODZUYzBMS3VsMnBlcnNvbmFsIn0.YnnMv2ApNpKPdh6aBCQm9hlB-W7C0DDyuzAMO2g61IA
I/flutter (29491): data:
I/flutter (29491): null
I/flutter (29491):
I/flutter (29491): *** Response ***
I/flutter (29491): uri: https://api.stream-io-api.com/api/v1.0/enrich/feed/timeline/0AJZfUOTqoXL8yByXY86Tc0LKul2personal/?limit=100&offset=0&api_key=hmwext26344q&location=unspecified
I/flutter (29491): statusCode: 200
I/flutter (29491): headers:
I/flutter (29491):  connection: keep-alive
I/flutter (29491):  cache-control: no-cache
I/flutter (29491):  x-ratelimit-reset: 1620850860
I/flutter (29491):  date: Wed, 12 May 2021 20:20:37 GMT
I/flutter (29491):  access-control-allow-origin: *
I/flutter (29491):  x-ratelimit-limit: 2000
I/flutter (29491):  content-encoding: gzip
I/flutter (29491):  x-ratelimit-remaining: 1997
I/flutter (29491):  content-type: application/json;charset=utf-8
I/flutter (29491):  access-control-max-age: 86400
I/flutter (29491):  server: nginx
I/flutter (29491):  access-control-allow-headers: x-requested-with, content-type, accept, origin, authorization, x-csrftoken, x-stream-client, stream-auth-type
I/flutter (29491):  access-control-allow-methods: GET, POST, PUT, PATCH, DELETE, OPTIONS
I/flutter (29491):  content-length: 471
I/flutter (29491): Response Text:
I/flutter (29491): {"results":[{"actor":{"created_at":"2021-04-13T22:53:19.670051Z","updated_at":"2021-04-13T22:53:19.670051Z","id":"eTHVBnEm0FQB2HeaRKVlEfVf58B3personal","data":{"gender":"Male","name":"Rickey Lee","photo":"https://firebasestorage.googleapis.com/v0/b/fire-snab.appspot.com/o/profile-image-placeholder.png?alt=media&token=b17598bb-a510-4167-8354-ab75642ba89e"}},"files":null,"foreign_id":"0853b531-a551-4333-a06c-459208c480af","hasFile":false,"id":"bc1118f0-b2d2-11eb-8c76-1254437355c5","object":"2021-05-11 23:32:53.044977","origin":"user:eTHVBnEm0FQB2HeaRKVlEfVf58B3personal","target":"","text":"Testing","time":"2021-05-12T03:32:53.304549","toExpire":false,"verb":"post"}],"next":"","duration":"10.80ms"}
```

**Comment by sachaarbonel** [comment]
did you add the activity with `addActivity(post)`?

**Comment by rlee1990** [comment]
@sachaarbonel  this is how I added the activity
```
final activity = feed.Activity(
                                      actor: feed.createUserReference(
                                          streamClient.currentUser.userId),
                                      verb: 'post',
                                      object: '${DateTime.now()}',
                                      foreignId: '${uuid.v4()}',
                                      time: DateTime.now().toLocal(),
                                      extraData: {
                                        'text': _postTextController.text.trim(),
                                        'files': uploadFiles,
                                        'hasFile': _hasFiles,
                                        'toExpire': isExpired
                                      });
                                  final userFeed = streamClient.flatFeed(
                                      'user', c.data?.value?.uid);
                                  userFeed.addActivity(activity).then((value) {
                                    WidgetsBinding
                                        .instance.focusManager.primaryFocus
                                        ?.unfocus();
                                    _postTextController.clear();
                                    AutoRouter.of(context).pop();
                                  });
```

**Comment by rlee1990** [comment]
The activity was created last night and the reaction was added today @sachaarbonel 
I am adding the reaction as a second user who follows the first user.

**Comment by sachaarbonel** [comment]
I'll investigate your logs from our dashboard but the first thing that seems odd to me is that you are overriding the foreignId, can you try not to? Also, not relevant but an object is something like a "tweet", a "photo" or a "video" for example: “Geraldine(actor) posted(verb) a photo(object) to her album(target)".

**Comment by rlee1990** [comment]
okay thanks for that I will make a new post this way and try it out. Also as an FYI I set all of that the same way as I have now when I used firebase functions to build everything and it worked without issue if that helps. I did see my logs on the dashboard on my end it shows green for post when adding the reactions but you probably can see more than I can. Thanks for the help.

**Comment by rlee1990** [comment]
@sachaarbonel here is a new post that I made following what you told me and then I added a reaction. I still get the same results: 
```
flutter: *** Request ***
flutter: uri: https://api.stream-io-api.com/api/v1.0/enrich/feed/timeline/eTHVBnEm0FQB2HeaRKVlEfVf58B3personal/?limit=100&offset=0&api_key=hmwext26344q&location=unspecified
flutter: method: GET
flutter: responseType: ResponseType.json
flutter: followRedirects: true
flutter: connectTimeout: 10000
flutter: sendTimeout: 0
flutter: receiveTimeout: 10000
flutter: receiveDataWhenStatusError: true
flutter: extra: {}
flutter: headers:
flutter:  stream-auth-type: jwt
flutter:  x-stream-client: stream-feed-dart-client-ios-0.1.2
flutter:  Authorization: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX2lkIjoiZVRIVkJuRW0wRlFCMkhlYVJLVmxFZlZmNThCM3BlcnNvbmFsIn0.IfL0sFc7vQUuQVBYDtJYp8luwOC0wQRWQ2znDSYlGds
flutter: data:
flutter: null
flutter:
flutter: *** Response ***
flutter: uri: https://api.stream-io-api.com/api/v1.0/enrich/feed/timeline/eTHVBnEm0FQB2HeaRKVlEfVf58B3personal/?limit=100&offset=0&api_key=hmwext26344q&location=unspecified
flutter: statusCode: 200
flutter: headers:
flutter:  connection: keep-alive
flutter:  cache-control: no-cache
flutter:  x-ratelimit-reset: 1620854280
flutter:  date: Wed, 12 May 2021 21:17:37 GMT
flutter:  access-control-allow-origin: *
flutter:  x-ratelimit-limit: 2000
flutter:  content-encoding: gzip
flutter:  x-ratelimit-remaining: 1999
flutter:  content-type: application/json;charset=utf-8
flutter:  access-control-max-age: 86400
flutter:  server: nginx
flutter:  access-control-allow-headers: x-requested-with, content-type, accept, origin, authorization, x-csrftoken, x-stream-client, stream-auth-type
flutter:  access-control-allow-methods: GET, POST, PUT, PATCH, DELETE, OPTIONS
flutter:  content-length: 557
flutter: Response Text:
flutter: {"results":[{"actor":{"created_at":"2021-04-13T23:04:05.306176Z","updated_at":"2021-04-13T23:04:05.306176Z","id":"0AJZfUOTqoXL8yByXY86Tc0LKul2personal","data":{"gender":"Female","name":"jane doe","photo":"https://firebasestorage.googleapis.com/v0/b/fire-snab.appspot.com/o/profile-image-placeholder.png?alt=media&token=b17598bb-a510-4167-8354-ab75642ba89e"}},"files":null,"foreign_id":"","hasFile":false,"id":"80334b08-b344-11eb-8080-80011019acc5","object":"texted","origin":"user:0AJZfUOTqoXL8yByXY86Tc0LKul2personal","target":"","text":"This is just a post again","time":"2021-05-12T17:07:15.492532","toExpire":false,"verb":"posted"},{"actor":{"created_at":"2021-04-13T23:04:05.306176Z","updated_at":"2021-04-13T23:04:05.306176Z","id":"0AJZfUOTqoXL8yByXY86Tc0LKul2personal","data":{"gender":"Female","name":"jane doe","photo":"https://firebasestorage.googleapis.com/v0/b/fire-snab.appspot.com/o/profile-image-placeholder.png?alt=media&token=b17598bb-a510-4167-8354-ab75642ba89e"}},"files":null,"for<…>
flutter:
flutter: EnrichedActivity(EnrichableField({created_at: 2021-04-13T23:04:05.306176Z, updated_at: 2021-04-13T23:04:05.306176Z, id: 0AJZfUOTqoXL8yByXY86Tc0LKul2personal, data: {gender: Female, name: jane doe, photo: https://firebasestorage.googleapis.com/v0/b/fire-snab.appspot.com/o/profile-image-placeholder.png?alt=media&token=b17598bb-a510-4167-8354-ab75642ba89e}}), EnrichableField(2021-05-12 12:34:37.432602), post, EnrichableField(), null, 4fcc86a7-1c8a-4da4-bf2d-8efb5083a545, 6a09d138-b31e-11eb-8080-8001022dd586, 2021-05-12 12:34:37.435116, null, null, EnrichableField(user:0AJZfUOTqoXL8yByXY86Tc0LKul2personal), null, {files: null, hasFile: false, text: This is just a test post., toExpire: false}, null, null, null)
```

**Comment by rlee1990** [comment]
Hi @sachaarbonel any update on this?

**Comment by sachaarbonel** [comment]
I tried to reproduce with a minimal sample and it's definitely a bug, working on a fix

**Comment by rlee1990** [comment]
@sachaarbonel Thanks for the update

**Comment by sachaarbonel** [comment]
@rlee1990 can you check if #62 solved your issue?

**Comment by rlee1990** [comment]
@sachaarbonel checking now.

**Comment by rlee1990** [comment]
@sachaarbonel I don't see any changes in the sdk and when I ran the app I am still not seeing anything.

**Comment by xsahil03x** [comment]
@rlee1990 change your pub dependency to git for testing the fix.
```dart
dependencies:   
  stream_feed:
     git:
       url: https://github.com/GetStream/stream-feed-flutter.git
       ref: fix-get-enriched-activities
       path: ./packages/stream_feed
```

**Comment by rlee1990** [comment]
@sachaarbonel I am not seeing like information I will make a new post and see what happens.

**Comment by rlee1990** [comment]
@sachaarbonel Sorry I thought I added a reply again but it is working. Thank you.

**Comment by sachaarbonel** [comment]
Hey @rlee1990 the fix is now available in  v0.1.3: `stream_feed: ^0.1.3`

**Comment by rlee1990** [comment]
@sachaarbonel  thanks.
